### PR TITLE
fix(48759): import<T> is valid syntax in nightly

### LINKED ISSUE #48759: `import<T>` is valid syntax in nightly
# Bug Report

Context: I'm trying to figure out how to update esbuild to handle the latest ASI-related behavior changes that just landed (#48755 and #48659). In the process, I discovered this weird TypeScript behavior that I believe was an oversight. I assume #47607 introduced the regression since it landed around the same time.

I tried to dig through code history to understand this better. It looks like `isStartOfLeftHandSideExpression` allows `<` after `import` due to #16544, although no tests were added for `import` followed by `<` in that PR. So I'm assuming TypeScript only allows `<` after `import` for parser recovery, which I don't need to support in esbuild.

### 🔎 Search Terms

regression import expression type parameters missing arguments

### 🕗 Version & Regression Information

- This changed between versions `4.7.0-dev.20220216` and `4.7.0-dev.20220217`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220418#code/JYWwDg9gTgLgPAFQHwG4BQAbApjABAMwglwF5dRJY5CIAaXAIwEMpU0g)

### 💻 Code

```ts
import<T>;
let foo = import<foo, bar>;
```

### 🙁 Actual behavior

The nightly TypeScript compiler converts the above TypeScript code into the following JavaScript without any errors:

```js
"use strict";
import;
let foo = (import);
```

### 🙂 Expected behavior

I either expected the TypeScript compiler to consider this invalid syntax, or if it's valid syntax I expected it to give type errors for the missing type names.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #48759

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -43862,6 +43862,9 @@ namespace ts {
         }
 
         function checkGrammarExpressionWithTypeArguments(node: ExpressionWithTypeArguments | TypeQueryNode) {
+            if (isExpressionWithTypeArguments(node) && isImportKeyword(node.expression) && node.typeArguments) {
+                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_cannot_have_type_arguments);
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:43866]: Is this the right error message and fix? It seems like `import` shouldn't be parsed as an `ExpressionWithTypeArguments` at all.

Comment (a-tarasyuk) [reply src/compiler/checker.ts:43866]: > Is this the right error message 

TS uses this error message for the following cases 
https://github.com/microsoft/TypeScript/blob/3d3fa0c0c83050d26390ef5773c471e15c8f1b21/tests/cases/conformance/dynamicImport/importCallExpressionWithTypeArgument.ts#L10-L11

TS throws the error in `checker`, I did the same for `ExpressionWithTypeArguments`

> It seems like import shouldn't be parsed as an ExpressionWithTypeArguments at all.

How should TypeScript parse `import <T>`? Without `()` this is not `ImportCallExpression`. I'm treating `import<T>` as a pre-`ImportCallExpression`, and I expect TypeScript to handle this and throw an error.

/cc @DanielRosenwasser @ahejlsberg

Comment (ahejlsberg) [reply src/compiler/checker.ts:43866]: I think this is the right approach. We want the parser to create a node that we can later issue a grammar error on, and `ExpressionWithTypeArguments` exactly fits the bill.

**Review by ahejlsberg — APPROVED** [review]

File: src/compiler/checker.ts (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/checker.ts:43866]: At the very least, we can't tell a user that this is a dynamic import. How about a new message that covers both.

> This use of 'import' is invalid. 'import()' calls can be written, but they must have parentheses and cannot have type arguments.

Comment (a-tarasyuk) [reply src/compiler/checker.ts:43866]: @DanielRosenwasser Oke.

**Pull request merged**